### PR TITLE
Fix FirewallRules concurency issues using a status known LastTransiti…

### PR DESCRIPTION
**Pull request by vincentmrg**
If a rule LastTransitionTime is before the one stored locally, the cache is not yet aware of a change we made. 
To avoid consistency issues, e.g. creating multiples groups for a single node, we requeue.

### CODE REVIEW

File: controllers/firewallrule_controller.go [hunk 1]
```diff
@@ -120,9 +142,18 @@ func (r *FirewallRuleReconciler) reconcileFirewallRule(ctx context.Context, log
 
 			// Check for other rules associated to the node.
 			// If there is already one, we update the group of rules, if not, we create a new group.
-			rulesAssociated := v1alpha1.FilterFirewallRules(frs.Items, func(fr v1alpha1.FirewallRule) bool {
-				return fr.Name != rule.Name && fr.Status.State != v1alpha1.FirewallRuleStateNone
-			})
+			rulesAssociated := []v1alpha1.FirewallRule{}
```
Comment (antonincms) [inline controllers/firewallrule_controller.go:145]: Is there a reason why you don't filter rules anymore ?

File: controllers/firewallrule_controller.go [hunk 2]
```diff
@@ -403,12 +472,14 @@ func (r *FirewallRuleReconciler) clearFirewallRule(ctx context.Context, log logr
 	}
 
 	// Update status
+	rule.Status.LastTransitionTime = metav1.Now()
 	rule.Status = v1alpha1.FirewallRuleStatus{State: v1alpha1.FirewallRuleStateNone}
```
Comment (antonincms) [inline controllers/firewallrule_controller.go:476]: That does not seem ok, you are setting LastTransitionTime and then overwriting it with an empty status.

Comment (vincentmrg) [reply controllers/firewallrule_controller.go]: Well seen

**Review by antonincms — APPROVED** [review]
Look good to me 👍

**Pull request merged**